File: 16-3sum-closest/3sum-closest.cpp
class Solution {

public:
    int threeSumClosest(vector<int>& nums, int target) {
        int n = nums.size();
        sort(nums.begin(),nums.end());
        
        int result=0;
        int min_diff = INT_MAX;
        for(int i=0;i<n-2;i++)
        {
            int left = i+1;
            int right = n-1;
            while(left<right)
            {
                int sum = nums[i]+nums[left]+nums[right];
                int diff = abs(sum-target);
                if(diff<min_diff)
                {
                    min_diff=diff;
                    result=sum;
                }
                if(sum<target) 
                left++;
                else if (sum>target)
                 right--;
                else {
                    
                    return sum;
                }
            }
        }
        return result;
    }
};
        

File: 16-3sum-closest/3sum-closest.py
class Solution(object):
    def threeSumClosest(self, nums, target):
        if not nums:
            return 0

        nums.sort()
        res = float('inf')
        temp = 0

        for i in range(len(nums) - 2):
            l = i + 1
            r = len(nums) - 1

            while l < r:
                total = nums[i] + nums[l] + nums[r]

                if abs(target - total) < res:
                    res = abs(target - total)
                    temp = total

                if total < target:
                    l += 1

                elif total > target:
                    r -= 1

                else:
                    return total

        return temp